db/compaction_iterator.h

```diff
@@ -21,6 +21,38 @@
 
 namespace rocksdb {
 
+class SnapshotListFetchCallback {
+ public:
+  SnapshotListFetchCallback(uint64_t _snap_refresh_nanos,
+                            size_t _every_nth_key = 1024)
+      : snap_refresh_nanos_(_snap_refresh_nanos),
+        every_nth_key_(_every_nth_key) {
+    assert(every_nth_key_ > 0);
+    assert((ceil(log2(every_nth_key_)) == floor(log2(every_nth_key_))));
+  }
+  virtual void Refresh(std::vector<SequenceNumber>* snapshots,
+                       SequenceNumber max) = 0;
+  inline bool TimeToRefresh(StopWatchNano& timer, size_t key_index,
+                            size_t snap_refresh_cnt) {
+    // skip the key if key_index % every_nth_key is not 0.
+    if ((key_index & (every_nth_key_ - 1)) != 0) {
```

Comment:
If we only use it in the form of `every_nth_key_ - 1`, why don't we store it instead?